src/common/src/types/mod.rs

```diff
@@ -341,8 +341,11 @@ impl DataType {
         };
         match self {
             DataType::Struct(t) => {
+                if !t.is_unnamed() {
+                    // To be consistent with `from_protobuf`, we only set field names when it's a named struct.
+                    pb.field_names = t.names().map(|s| s.into()).collect();
+                }
                 pb.field_type = t.types().map(|f| f.to_protobuf()).collect();
-                pb.field_names = t.names().map(|s| s.into()).collect();
```

Comment:
before this PR, `names()` returns empty, right?.

Reply:
Right